restapi.module

```diff
@@ -729,3 +730,42 @@ function _restapi_get_versioned_method(ResourceConfigurationInterface $resource,
 
   return method_exists($resource->getClass(), $method) ? $method : NULL;
 }
+
+
+/**
+ * Helper function to determine whether the response is a file download.
+ *
+ * @param ResponseInterface $response
+ *   The response.
+ *
+ * @return boolean
+ *   Returns TRUE when the response is a download, FALSE otherwise.
+ *
+ */
+function _restapi_is_download_response(ResponseInterface $response) {
+
+  $is_download = FALSE;
+
+  // If this content transfer encoding is a binary file, then this is a download.
+  $content_encoding = array_map('strtolower', $response->getHeader('Content-Transfer-Encoding'));
+  if (in_array('binary', $content_encoding)) {
+    $is_download = TRUE;
+  }
+
+  // If content disposition specifies an attachment, then this is a download.
+  $content_disposition = array_filter($response->getHeader('Content-Disposition'), function ($header) {
```

Comment:
This should be the [only test that matters](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Disposition): if it's attachment, it's a download. Anything else should be displayed in the browser.